parl/agent_zoo/agent_helpers.py

```diff
@@ -0,0 +1,145 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+from threading import Thread, Lock
+from parl.common.error_handling import check_type_error
+from parl.common.replay_buffer import NoReplacementQueue, ReplayBuffer
+from parl.framework.agent import AgentHelperBase, DataProcessorBase
+
+
+class RLDataProcessor(DataProcessorBase):
+    def __init__(self, specs):
+        super(RLDataProcessor, self).__init__(specs)
+
+    def process_prediction_inputs(self, inputs, states):
+        assert bool(inputs)
+        new_inputs = {}
+        for k in inputs:
+            new_inputs[k] = np.stack([inputs[k]])
+        data = dict(inputs=new_inputs)
+        if bool(states):
+            new_states = {}
+            for k in inputs:
+                new_states[k] = np.stack([new_states[k]])
+            data["states"] = new_states
+        return data
+
+    def process_learning_inputs(self, exp_seqs):
+        def append_data(data, attr, exp_seq):
+            if attr.startswith("next_"):
+                l = [getattr(exp.next_exp, attr[5:]) for exp in exp_seq]
```

Comment:
maybe could add some comments for the structure of attr for help understanding attr[5:]?